pkg/alerts/alertsHandler/alertsHandler.go

```diff
@@ -173,6 +181,115 @@ func ProcessSilenceAlertRequest(ctx *fasthttp.RequestCtx) {
 	utils.WriteJsonResponse(ctx, responseBody)
 }
 
+func ProcessTestContactPointRequest(ctx *fasthttp.RequestCtx) {
+	var testContactRequest TestContactPointRequest
+	if err := json.Unmarshal(ctx.PostBody(), &testContactRequest); err != nil {
+		utils.SendError(ctx, "Failed to unmarshal json", "ProcessTestContactPointRequest: Request Body: "+string(ctx.PostBody()), err)
+		return
+	}
+
+	switch testContactRequest.Type {
+	case "slack":
+		channelID, ok := testContactRequest.Settings["channel_id"].(string)
+		if !ok {
+			utils.SendError(ctx, "channel_id is required but is missing", "ProcessTestContactPointRequest: Slack Channel ID is missing. Request Body: "+string(ctx.PostBody()), nil)
+			return
+		}
+		slackToken, ok := testContactRequest.Settings["slack_token"].(string)
+		if !ok {
+			utils.SendError(ctx, "slack_token is required but is missing", "ProcessTestContactPointRequest: Slack token is missing. Request Body: "+string(ctx.PostBody()), nil)
+			return
+		}
+		err := sendTestMessageToSlack(channelID, slackToken)
+		if err != nil {
+			utils.SendError(ctx, err.Error(), "ProcessTestContactPointRequest: Error sending test message to slack. Request Body:"+string(ctx.PostBody()), err)
+			return
+		}
+	case "webhook":
+		webhookURL, ok := testContactRequest.Settings["webhook"].(string)
+		if !ok {
+			utils.SendError(ctx, "webhook is required but is missing", "ProcessTestContactPointRequest: Webhook URL is missing Request Body: "+string(ctx.PostBody()), nil)
+			return
+		}
+		if err := testWebhookURL(webhookURL); err != nil {
+			utils.SendError(ctx, "Failed to verify webhook URL", "", err)
+			return
+		}
+	default:
+		utils.SendError(ctx, "Invalid type", "ProcessTestContactPointRequest: Received unsupported test contact request type Request Body:"+string(ctx.PostBody()), nil)
+		return
+	}
+
+	ctx.SetStatusCode(fasthttp.StatusOK)
+	utils.WriteJsonResponse(ctx, map[string]interface{}{"message": "Successfully verified contact point"})
+}
+
+func sendTestMessageToSlack(channelID, slackToken string) error {
```

Comment:
Also, I think it is better if we use the existing `sendSlack` function or the `slack` pkg, instead of sending this through this http post message, as that pkg is what we are using to actually send the Alert messages.

You could probably use the `sendSlack` method or mimic something similar but use the `slack` pkg. Check these lines of code: https://github.com/siglens/siglens/blob/develop/pkg/alerts/alertsHandler/notificationHandler.go#L195-L219

@AndrewHess @nkunal Thoughts on this?

Reply:
I agree. I think you can just call the `sendSlack` function. But if there's a reason not to, then you could use the `slack` package like Mani said and it would probably help clean up the logic